File: code_samples/elixir/security_hashing.ex
# Security: Hashing
# AI/ML Training Sample

defmodule Hashing do
  defstruct data: ""
  
  def new(), do: %Hashing{}
  
  def process(%Hashing{} = struct, input) do
    %{struct | data: input}
  end
  
  def get_data(%Hashing{data: data}), do: data
  
  def validate(%Hashing{data: data}) do
    String.length(data) > 0
  end
end

# Example usage
instance = Hashing.new()
updated = Hashing.process(instance, "example")
IO.puts("Data: " <> Hashing.get_data(updated))
IO.puts("Valid: " <> to_string(Hashing.validate(updated)))

File: code_samples/typescript/security_hashing.ts
/**
 * Security: Hashing
 * AI/ML Training Sample
 */

interface IHashing {
    data: string;
    process(input: string): void;
    validate(): boolean;
}

class Hashing implements IHashing {
    data: string;
    
    constructor() {
        this.data = "";
    }
    
    process(input: string): void {
        this.data = input;
    }
    
    getData(): string {
        return this.data;
    }
    
    validate(): boolean {
        return this.data.length > 0;
    }
}

// Example usage
const instance = new Hashing();
instance.process("example");
console.log(`Data: ${instance.getData()}`);
console.log(`Valid: ${instance.validate()}`);

export { Hashing, IHashing };
